framework/CodeInterfaces/MooseBasedApp/MOOSEparser.py

```diff
@@ -23,258 +23,103 @@
-          print("Missmatch ",key,repr(dictionary[key]),repr(other[key]))
-          returnBool = False
-      else:
-        binKey = toBytes(key)
-        if binKey in dictionary:
-          if not compare(dictionary[binKey],other[key]):
-            print("Missmatch_b ",key,dictionary[binKey],other[key])
-            returnBool = False
-        else:
-          print("No_key ",key,other[key])
-          returnBool = False
-    return returnBool
+    # if asking to erase a block ...
+    # TODO is this untested code?
+    if 'erase_block' in specials:
+      if found:
+        found[-2].remove(found[-1])
 
-  def __modifyOrAdd(self,returnElement,name,modiDictionary):
-    """
-      Modify the XML tree with the information in name and modiDictionary
-      If erase_block in modiDictionary, then remove name from returnElement
-      else modify name in returnElement
-      @ In, returnElement, xml.etree.ElementTree.Element, the tree that needs to be updated
-      @ In, name, list, list of instruction to reach the node to be modified
-      @ In, modiDictionary, dict, dictionary contained the info to modify the tree
-      @ Out, None
-    """
-    assert(len(name) > 0)
-    specials  = modiDictionary['special'] if 'special' in modiDictionary.keys() else set()
-    #If erase_block is true, then erase the entire block
-    has_erase_block = 'erase_block' in specials
-    #If assert_match is true, then fail if any of the elements do not exist
-    has_assert_match = 'assert_match' in specials
-    #If name[0] is not found and in erase_block, then done
-    found,true_name = self.__findInXML(returnElement,name[0])
-    if not found and has_erase_block:
-      #Not found, and just wanted to erase it, so quit.
-      return
-    if not found and has_assert_match:
-      #Not found, and just checking to see if there was a match
-      return
-    #If len(name) == 1, then don't recurse anymore.  Either
-    # erase block or modify the element.
-    if len(name) == 1:
-      modiDictionary.pop('special',None)
-      if has_erase_block:
-        returnElement.remove(returnElement.find(true_name))
-      elif has_assert_match:
-        self.__matchDict(returnElement.find(true_name).attrib,modiDictionary)
-        assert(self.__matchDict(returnElement.find(true_name).attrib,modiDictionary))
-      elif found:
-        self.__updateDict(returnElement.find(true_name).attrib,modiDictionary)
-      else:
-        ET.SubElement(returnElement,name[0],modiDictionary)
-    else:
-      if not found:
-        subElement = ET.SubElement(returnElement,name[0])
-        #if len(name) > 1, then if not found (and since we already checked for erasing) then add it and recurse.
-      else:
-        # if len(name) > 1 and found, then recurse on child
-        subElement = returnElement.find(true_name)
-      self.__modifyOrAdd(subElement,name[1:],modiDictionary)
+    # if asking to check on a block ...
+    # TODO is this untested code?
```

Comment:
it looks like it is not tested...it was with Grizzly but we probably lost that test